### PR TITLE
refactor: icon to icons in DatasourceEditor

### PR DESCRIPTION
**Pull request by pkdotson**
### SUMMARY
This pr uses the new Icons component for the datasourceeditor.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
before:
Screen Shot 2021-06-17 at 2 43 06 PM

after
Screen Shot 2021-06-17 at 2 43 18 PM

### TESTING INSTRUCTIONS
Go to datasource editor modal and check lock icon.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/datasource/DatasourceEditor.jsx [hunk 1]
```diff
@@ -94,6 +94,9 @@ const EditLockContainer = styled.div`
   a {
     padding: 0 10px;
   }
+  span[role='img'] {
+    margin-top: -3px;
+  }
```
Comment (rusackas) [inline superset-frontend/src/datasource/DatasourceEditor.jsx:99]:
```suggestion
```
I would just take this out. It looks fine without.

**Review by rusackas — APPROVED** [review]
LGTM. Just committed one lil' tweak.

File: superset-frontend/src/datasource/DatasourceEditor.jsx [hunk 2]
```diff
@@ -842,10 +842,15 @@ class DatasourceEditor extends React.PureComponent {
         {this.allowEditSource && (
           <EditLockContainer>
             <span role="button" tabIndex={0} onClick={this.onChangeEditMode}>
-              <Icon
-                color={supersetTheme.colors.grayscale.base}
-                name={this.state.isEditMode ? 'lock-unlocked' : 'lock-locked'}
-              />
+              {this.state.isEditMode ? (
+                <Icons.LockUnlocked
+                  iconColor={supersetTheme.colors.grayscale.base}
```
Comment (michael-s-molina) [inline superset-frontend/src/datasource/DatasourceEditor.jsx:847]: Can you change to use `withTheme` or `useTheme` to reference the `theme` object?

**Review by michael-s-molina — CHANGES_REQUESTED** [review]

File: superset-frontend/src/datasource/DatasourceEditor.jsx (hunk 2 above)
Comment (pkdotson) [reply superset-frontend/src/datasource/DatasourceEditor.jsx:847]: I'm going to do this in a follow up and get rid of all the supersetThemes being used in components.

**Review by michael-s-molina — APPROVED** [review]

**Pull request merged**